src/main/java/org/uast/astgen/utils/FilesWriter.java

```diff
@@ -36,7 +36,7 @@ public FilesWriter(final String path) {
      * @throws IOException If the file can't be written
      */
     public void writeString(final String str) throws IOException {
-        Files.createDirectories(this.path.subpath(0, this.path.getNameCount() - 1));
+        Files.createDirectories(this.path.getParent());
```

Comment:
```
        final Path parent = this.path.getParent();
        if (parent != null) {
            Files.createDirectories(this.path.getParent());
        }
```
In your case, the program crashes if the path contains only name of file, without folder names.

Reply:
How? I ran it without option `--output` and it worked because we have a default directory.